openproblems/tasks/spatial_decomposition/datasets/_utils.py

```diff
@@ -0,0 +1,88 @@
+from .._utils import obs_means
+from anndata import AnnData
+
+import numpy as np
+import random
+
+
+def generate_seq(length=14):
+    return "".join(random.choice("CGTA") for _ in range(length))
+
+
+# pass the reference data
+def generate_synthatic_data(adata, sim_type="avg"):
+    n_genes = adata.n_var
+    n_cells = adata.n_obs
+    n_types = len(set(adata.obs["label"].values))
+
+    # TODO(make these arguments)
+    bead_depth = 1000
```

Comment:
Maybe `bead_depth` could be made variable, as to not have more of a heterogenous distribution of cells? We could sample from a uniform (integers) distribution with a lower and upper bound perhaps?

Reply:
yes, I thought about that I will make a TODO for the same.